### PR TITLE
ISC Restructure - Small Fixes

### PR DESCRIPTION
**Pull request by lucas-tortora**
# Description of change

- [ ] Danger to info in languages and vms
- [ ] Review note in intro about on and off ledger requests, add link to actual explanation
- [ ] double check ShimmerEVM Testnet nomenclature
- [ ] Move test smart contracts down, add examples
- [ ] add link to deploy in basic SC
- [ ] remove highlight ERC721
- [ ] move EVM to top layer (AKA group by intent or topic, rather than VM)
- [ ] rotate full example where it's different
- [ ] tabs in create foundry
- [ ] tabs in mint native tokens
- [ ] review intructions in register-token
- [ ] review headers allowance
- [ ] add warning about ERC20 in getbalance
- [ ] add note about onledger requests on /isc/explanations/invocation/
- [ ] move /isc/explanations/proxies/ to wasm
- [ ] add link to JSON RPC in isc/getting-started/compatibility/#wrapped-calls-to-the-json-rpc
## Links to any relevant issues

fixes https://github.com/iotaledger/devx/issues/341

## Type of change

- Documentation Fix

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: docs/build/isc/v1.0.0-rc.6/docs/_admonitions/_off-ledger-requests.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+:::note
+
+If you are using EVM, you will only need to use off-ledger requests for most of your contracts.
```
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/_admonitions/_off-ledger-requests.md:3]: only and most seem to communicate different things 😅 
But I would rephrase this in general. As you can't really use on-ledger requests for anything outside magic or core contracts

File: docs/build/isc/v1.0.0-rc.6/docs/_admonitions/_off-ledger-requests.md
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/_admonitions/_off-ledger-requests.md:1]: only and most seem to communicate different things 😅
But I would rephrase this in general. As you can't really use on-ledger requests for anything outside magic or core contracts

File: docs/build/isc/v1.0.0-rc.6/docs/explanations/invocation.md
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/explanations/invocation.md:1]: Does this only look fucked up in my preview?

File: docs/build/isc/v1.0.0-rc.6/docs/getting-started/compatibility.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ Here are some of the most important properties and limitations of EVM support in
 
 ### Wrapped Calls to the JSON-RPC
 
-The Wasp node provides a JSON-RPC service, the standard protocol used by Ethereum tools. Upon receiving a signed
+The Wasp node provides a [JSON-RPC](../reference/json-rpc-spec.md) service, the standard protocol used by Ethereum tools. Upon receiving a signed
```
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/getting-started/compatibility.md:36]: We should not only link to it, but only mention that it lists the implemented/supported endpoints. Btw we should update the status there. What even does the hour clock mean? 😆 
Maybe something like this?
```suggestion
The Wasp node provides a JSON-RPC service, the standard protocol used by Ethereum tools. We try to keep our endpoint as compatible as possible, you can find a list of supported endpoints [here](../reference/json-rpc-spec.md). Upon receiving a signed
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/get-balance.md [hunk 3]
```diff
@@ -7,11 +7,14 @@ tags:
   - EVM
   - how-to
 ---
+import LayerBalanceDiff from '../_admonitions/_layer_balance_diff.md';
```
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/get-balance.md:10]:
```suggestion
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/get-balance.md [hunk 4]
```diff
@@ -7,11 +7,14 @@ tags:
...
 
 # Get Balance
 
 Once you have your L1 assets on L2, you might want to check their balance. This guide explains how to do so by calling the three functions `getL2BalanceBaseTokens`, `getL2BalanceNativeTokens`and `getL2NFTAmount` for the corresponding token types.
 
+<LayerBalanceDiff />
+
```
Comment (Dr-Electron) [inline docs/build/isc/v1.0.0-rc.6/docs/how-tos/get-balance.md:17]:
```suggestion
```

File: docs/build/isc/v1.0.0-rc.6/docs/how-tos/get-balance.md (hunk 3 above)
Comment (Dr-Electron) [reply docs/build/isc/v1.0.0-rc.6/docs/how-tos/get-balance.md:10]: We don't need it here. I will push a fix later

**Review by Dr-Electron — CHANGES_REQUESTED** [review]
I think the EVM change was not intended that way. @jlvandenhout thought more about a rename. 
But in general I think we should rethink that section

**Comment by Dr-Electron** [comment]
I would restructure the sidebar like this:

From:
```
How-Tos
├── About How-to Guides
├── Send Funds From L1 to L2
├── Create a Basic Solidity Contract
├── Deploy a Smart Contract
├── Test Smart Contracts
├── Create Custom Tokens - ERC20
├── Create NFTs - ERC721
└── Use Magit Contract/
```
To:
```
How-Tos
├── About How-to Guides
├── Send Funds From L1 to L2
├── Create a Basic Contract // Make the title VM agnostic.
├── Deploy a Smart Contract
├── Test Smart Contracts // This is actually not really a how-to. More of an explanation I guess
└── Use Magit Contract/
```

**Comment by lucas-tortora** [comment]
no ERC.. examples?

> I would restructure the sidebar/content like this:
> 
> From:
> 
> ```
> How-Tos
> ├── About How-to Guides
> ├── Send Funds From L1 to L2
> ├── Create a Basic Solidity Contract
> ├── Deploy a Smart Contract
> ├── Test Smart Contracts
> ├── Create Custom Tokens - ERC20
> ├── Create NFTs - ERC721
> └── Use Magit Contract/
> ```
> 
> To:
> 
> ```
> How-Tos
> ├── About How-to Guides
> ├── Send Funds From L1 to L2
> ├── Create a Basic Contract // Make the title VM agnostic.
> ├── Deploy a Smart Contract
> ├── Test Smart Contracts // This is actually not really a how-to. More of an explanation I guess
> └── Use Magit Contract/
> ```

No Solidity ERC how tos?

**Comment by Dr-Electron** [comment]
> no ERC.. examples?
> 
> > I would restructure the sidebar/content like this:
> > From:
> > ```
> > How-Tos
> > ├── About How-to Guides
> > ├── Send Funds From L1 to L2
> > ├── Create a Basic Solidity Contract
> > ├── Deploy a Smart Contract
> > ├── Test Smart Contracts
> > ├── Create Custom Tokens - ERC20
> > ├── Create NFTs - ERC721
> > └── Use Magit Contract/
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > To:
> > ```
> > How-Tos
> > ├── About How-to Guides
> > ├── Send Funds From L1 to L2
> > ├── Create a Basic Contract // Make the title VM agnostic.
> > ├── Deploy a Smart Contract
> > ├── Test Smart Contracts // This is actually not really a how-to. More of an explanation I guess
> > └── Use Magit Contract/
> > ```
> 
> No Solidity ERC how tos?

No. As we once said there are tutorials others did better. Our USP is the core contracts basically

**Comment by lucas-tortora** [comment]
> > no ERC.. examples?
> > > I would restructure the sidebar/content like this:
> > > From:
> > > ```
> > > How-Tos
> > > ├── About How-to Guides
> > > ├── Send Funds From L1 to L2
> > > ├── Create a Basic Solidity Contract
> > > ├── Deploy a Smart Contract
> > > ├── Test Smart Contracts
> > > ├── Create Custom Tokens - ERC20
> > > ├── Create NFTs - ERC721
> > > └── Use Magit Contract/
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > To:
> > > ```
> > > How-Tos
> > > ├── About How-to Guides
> > > ├── Send Funds From L1 to L2
> > > ├── Create a Basic Contract // Make the title VM agnostic.
> > > ├── Deploy a Smart Contract
> > > ├── Test Smart Contracts // This is actually not really a how-to. More of an explanation I guess
> > > └── Use Magit Contract/
> > > ```
> > 
> > 
> > No Solidity ERC how tos?
> 
> No. As we once said there are tutorials others did better. Our USP is the core contracts basically

That's one way of looking at it. The other is that we should cover the basics. There's a middle ground which would be VM agnostic, but cover Solidity as well

**Review by Dr-Electron — APPROVED** [review]
